app.rb

```diff
@@ -0,0 +1,148 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    type = gets.chomp.to_i
+    case type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid input. Please enter 1 for a student or 2 for a teacher.'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp
+    person = Student.new(name, age, parent_permission)
+    @people.push(person)
+    puts "Student '#{name}' created successfully"
+  end
```

Comment:
- [ ] This method does not obey the Single responsibility principle. It is responsible for requesting user inputs and also creating the student object. Kindly move the request for user inputs to another class or file responsible for requesting user inputs.